datastore/mongodb/filter_outputs.go

```diff
@@ -31,47 +31,80 @@ func (c *Client) CreateFilterOutput(ctx context.Context, f *model.FilterOutput)
 func (c *Client) UpdateFilterOutput(ctx context.Context, f *model.FilterOutput) error {
 	col := c.collections.filterOutputs
 
-	var docs []model.FilterOutput
-	sc := bson.M{"id": f.ID}
-
-	var err error
-	var num int
-
-	if num, err = c.conn.Collection(col.name).Find(ctx, sc, &docs, mongodb.Limit(1)); err != nil {
-		return errors.Wrap(err, "failed to update filter output")
-	}
-	if num < 1 {
-		return &er{
-			err:      errors.Errorf("failed to find filter output"),
-			notFound: true,
-		}
+	docs, err := c.findFilterOutput(f.ID, ctx)
+	if err != nil {
+		return err
 	}
-
 	//a record with stat 'completed' can't be updated further
 	if docs[0].State == model.Completed {
-		return &er{
-			err:       errors.Errorf(`filter output is already in "completed" state`),
-			forbidden: true,
+		err := &er{
```

Comment:
What was wrong with how this was done before?

Reply:
It is just an outcome of bad merge